src/ai/backend/agent/docker/metadata/server.py

```diff
@@ -143,7 +143,7 @@ async def route_structure_fallback_middleware(
             structure_pointer = self.route_structure
             for component in components:
                 if structure_pointer.get(component) is None:
-                    raise web.HTTPNotFound
+                    raise web.HTTPNotFound from None
```

Comment:
In this case, I think we should simply reraise the original exception.
```suggestion
                    raise
```